Content.Shared/Markings/Marking.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using System.Collections.Generic;
+using Robust.Shared.Maths;
+using Robust.Shared.Serialization;
+using Robust.Shared.Serialization.Manager.Attributes;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Shared.Markings
+{
+    [Serializable, NetSerializable]
+    public sealed class Marking : IEquatable<Marking>, IComparable<Marking>, IComparable<string>
+    {
+        private List<Color> _markingColors = new();
+
+        private Marking(string markingId,
+            List<Color> markingColors)
+        {
+            MarkingId = markingId;
+            _markingColors = markingColors;
+        }
+
+        public Marking(string markingId,
+            IReadOnlyList<Color> markingColors)
+            : this(markingId, new List<Color>(markingColors))
+        {
+        }
+
+        /*
+        public Marking(string markingId)
+            : this(markingId, new List<Color>())
+        {
+        }
+        */
+
+        public Marking(string markingId, int colorCount)
+        {
+            MarkingId = markingId;
+            List<Color> colors = new();
+            for (int i = 0; i < colorCount; i++)
+                colors.Add(Color.White);
+            _markingColors = colors;
+        }
+
+        [DataField("markingId")]
+        [ViewVariables]
+        public string MarkingId { get; } = default!;
+
+        [DataField("markingColor")]
+        [ViewVariables]
+        public IReadOnlyList<Color> MarkingColors => _markingColors;
+
+        public void SetColor(int colorIndex, Color color) =>
+            _markingColors[colorIndex] = color;
+
+        public int CompareTo(Marking? marking)
+        {
+            if (marking == null) return 1;
+            else return this.MarkingId.CompareTo(marking.MarkingId);
+        }
+
+        public int CompareTo(string? markingId)
+        {
+            if (markingId == null) return 1;
+            return this.MarkingId.CompareTo(markingId);
+        }
+
+        public bool Equals(Marking? other)
+        {
+            if (other == null) return false;
+            return (this.MarkingId.Equals(other.MarkingId));
+        }
+
+        
+        // look this could be better but I don't think serializing
+        // colors is the correct thing to do
+        new public string ToString()
+        {
+            // reserved character
+            string sanitizedName = this.MarkingId.Replace('@', '_');
+            List<string> colorStringList = new();
+            foreach (Color color in _markingColors)
+                colorStringList.Add(color.ToHex());
+
+            return $"{sanitizedName}@{String.Join(',', colorStringList)}";
+        }
```

Comment:
sussy

Reply:
workaround for serialization, but i could just rename this / actually figure out serialization stuff (i.e., ColorSerializer) - currently it's just serializing straight to a JSON list because of the need of order preservation